### PR TITLE
Same branch on re-submit

### LINKED ISSUE #55: Same branch on re-submitted transcripts
If a transcript has been submitted earlier, then resubmitted after some changes, the same branch used to make the PR initially on create PR logic should be re-used. This keeps the further iterations of that transcript on the same PR.

### PR DESCRIPTION
**Pull request by 0tuedon**
This Pr closes #55

### CODE REVIEW

File: src/components/transcript/index.tsx [hunk 1]
```diff
@@ -216,6 +216,7 @@ const Transcript = ({ reviewData }: { reviewData: UserReviewData }) => {
         directoryPath: loc?.trim() ? loc.trim() : config.defaultDirectoryPath,
         fileName: formatDataForMetadata(title),
         url: transcriptData?.content.media,
+        prUrl: reviewData?.pr_url || undefined,
```
Comment (Emmanuel-Develops) [inline src/components/transcript/index.tsx:219]: why or undefined?

File: src/pages/api/github/pr.ts [hunk 2]
```diff
@@ -11,11 +11,11 @@ async function createForkAndPR(
   fileName: string,
   transcribedText: string,
   metaData: Metadata,
-  prRepo: TranscriptSubmitOptions
+  prRepo: TranscriptSubmitOptions,
+  prUrl: string
```
Comment (Emmanuel-Develops) [inline src/pages/api/github/pr.ts:15]: prUrl is optional

File: src/pages/api/github/pr.ts [hunk 3]
```diff
@@ -27,31 +27,56 @@ async function createForkAndPR(
   const forkOwner = forkResult.data.owner.login;
   const forkRepo = upstreamRepo;
 
+  const prUrlLength = prUrl ? prUrl.split("/").length : 1;
```
Comment (Emmanuel-Develops) [inline src/pages/api/github/pr.ts:30]: you can pull the pull_number from here to use in other parts of the code

File: src/pages/api/github/pr.ts [hunk 4]
```diff
@@ -27,31 +27,56 @@ async function createForkAndPR(
...
+  //  get's the pull request files and also their SHA if they exist
+  const pullFiles = await octokit
+    .request("GET /repos/{owner}/{repo}/pulls/{pull_number}/files", {
+      owner: forkOwner,
+      repo: forkRepo,
+      pull_number: prUrl ? +prUrl.split("/")[prUrlLength - 1] : 0,
+    })
+    .then((pr) => ({ data: pr?.data[0] }))
+    .catch((err) => err);
```
Comment (Emmanuel-Develops) [inline src/pages/api/github/pr.ts:39]: throw the err;
so that it can be caught in the handler;

File: src/pages/api/github/pr.ts [hunk 5]
```diff
@@ -27,31 +27,56 @@ async function createForkAndPR(
...
+  // get details about the PR also gets the branch name (so we are able to commit)
+  const pullDetails: any = await octokit
+    .request("GET /repos/:owner/:repo/pulls/:pull_number", {
+      owner: forkOwner,
+      repo: forkRepo,
+      pull_number: prUrl ? +prUrl.split("/")[prUrlLength - 1] : 0,
+    })
+    .then((_data) => _data)
+    .catch((err) => {
+      return err;
```
Comment (Emmanuel-Develops) [inline src/pages/api/github/pr.ts:50]: throw err; here too

**Review by Emmanuel-Develops — CHANGES_REQUESTED** [review]

File: src/components/transcript/index.tsx (hunk 1 above)
Comment (0tuedon) [reply src/components/transcript/index.tsx:219]: i see now that's it's a repition  reviewData?.pr_url would return undefined when there's no pr_url.

**Review by Emmanuel-Develops — APPROVED** [review]

**Review by BalogunofAfrica — APPROVED** [review]

File: src/pages/api/github/pr.ts [hunk 6]
```diff
@@ -108,28 +136,39 @@ async function createForkAndPR(
...
-    content: fileContent,
-    branch: newBranchName,
-  });
+  const updateContent = await octokit.request(
+    "PUT /repos/:owner/:repo/contents/:path",
+    {
+      owner: forkOwner,
+      repo: forkRepo,
+      path: `${directoryPath}/${fileSlug}.md`,
+      message: `${prUrl ? "Updated" : "Added"} "${
```
Comment (adamjonas) [inline src/pages/api/github/pr.ts:145]: Screenshot 2023-10-18 at 4 27 16 PM

The updated line is really gross with the "" "" next to each other

**Pull request merged**